subsys/bluetooth/audio/ascs.c

```diff
@@ -883,27 +883,32 @@ NET_BUF_SIMPLE_DEFINE_STATIC(ase_buf, CONFIG_BT_L2CAP_TX_MTU);
 static void ase_process(struct k_work *work)
 {
 	struct bt_ascs_ase *ase = CONTAINER_OF(work, struct bt_ascs_ase, work);
+	struct bt_audio_ep *ep = &ase->ep;
+	const struct bt_audio_iso *audio_iso = ep->iso;
+	struct bt_audio_stream *stream = ep->stream;
+	const uint8_t ep_state = ep->status.state;
 
-	BT_DBG("ase %p, ep %p, ep.stream %p", ase, &ase->ep, ase->ep.stream);
+	BT_DBG("ase %p, ep %p, ep.stream %p", ase, ep, stream);
 
 	if (ase->ascs->conn != NULL &&
 	    ase->ascs->conn->state == BT_CONN_CONNECTED) {
-		ascs_ep_get_status(&ase->ep, &ase_buf);
+		ascs_ep_get_status(ep, &ase_buf);
 
-		bt_gatt_notify(ase->ascs->conn, ase->ep.server.attr,
+		bt_gatt_notify(ase->ascs->conn, ep->server.attr,
 			       ase_buf.data, ase_buf.len);
 	}
 
-	if (ase->ep.status.state == BT_AUDIO_EP_STATE_RELEASING) {
-		struct bt_audio_stream *stream = ase->ep.stream;
+	/* Stream shall only be NULL in the idle state */
+	__ASSERT(ep_state == BT_AUDIO_EP_STATE_IDLE ||
+		 stream != NULL, "stream is NULL");
```

Comment:
I'd suggest to assert if stream is expected to be non-null as well 
`__ASSERT_NO_MSG(ep_state == BT_AUDIO_EP_STATE_IDLE ? stream == NULL : stream != NULL)`